com.sap.cloud.lm.sl.mta/src/main/java/com/sap/cloud/lm/sl/mta/resolvers/PropertiesResolver.java

```diff
@@ -108,17 +115,63 @@ protected Object resolveReferenceInContext(String key, Reference reference) {
     private Object resolveReference(Reference reference) {
         String referencedPropertyKey = reference.getPropertyName();
         Map<String, Object> replacementValues = valuesResolver.resolveProvidedValues(reference.getDependencyName());
-        if (replacementValues == null || !replacementValues.containsKey(referencedPropertyKey)) {
+
+        boolean canResolveInDepth = referencePattern.hasDepthOfReference() && referencedPropertyKey.contains("/");
+        if (replacementValues == null || (!replacementValues.containsKey(referencedPropertyKey) && !canResolveInDepth)) {            
             if (isStrict) {
                 throw new ContentException(Messages.UNABLE_TO_RESOLVE, getPrefixedName(prefix, referencedPropertyKey));
             }
             return null;
         }
-        String referencedPropertyKeyWithSuffix = getReferencedPropertyKeyWithSuffix(reference);
+        // always try to resolve as a flat reference first
         Object referencedProperty = replacementValues.get(referencedPropertyKey);
+        
+        if (referencedProperty == null && canResolveInDepth) {
+            referencedProperty = resolveInDepth(replacementValues, referencedPropertyKey);
+        }
+        
+        String referencedPropertyKeyWithSuffix = getReferencedPropertyKeyWithSuffix(reference);
         return resolve(referencedPropertyKeyWithSuffix, referencedProperty);
     }
-
+    
+    protected Object resolveInDepth(Map<String, Object> properties, String propertyKey) {

```

Comment:
I would rename `properties` to `replacementValues` here and `propertyKey` to `referencedPropertyKey`. It would be more consistent with the code above and is also IMO easier to understand. When I looked at `currentProperty` below I thought that this was the property that contained the reference and not the actual replacement value.